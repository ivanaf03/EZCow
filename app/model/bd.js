import * as SQLite from "expo-sqlite";

let db;

export const getDatabase = async () => {
  if (!db) {
    db = await SQLite.openDatabaseAsync("ezcow.db");
  }
  return db;
};

export const initDatabase = async () => {
  db = await getDatabase();

  await db.execAsync(`
        PRAGMA journal_mode = WAL;

        CREATE TABLE IF NOT EXISTS User (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            email TEXT NOT NULL UNIQUE,
            password TEXT DEFAULT NULL,
            google_id TEXT DEFAULT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            CHECK (password IS NOT NULL OR google_id IS NOT NULL)
        );

        CREATE TABLE IF NOT EXISTS Cow (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            code TEXT NOT NULL UNIQUE,
            name TEXT NOT NULL,
            entryDate DATE NOT NULL,
            exitDate DATE DEFAULT NULL,
            gender TEXT NOT NULL,
            breed TEXT NOT NULL,
            phase TEXT NOT NULL,
            user_fk INTEGER NOT NULL,
            mother_fk INTEGER DEFAULT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id),
            FOREIGN KEY(mother_fk) REFERENCES Cow(id)
        );

        CREATE TABLE IF NOT EXISTS Field (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            user_fk INTEGER NOT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id)
        );

        CREATE TABLE IF NOT EXISTS BreedingEvent (
            if INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            description TEXT,
            date DATE NOT NULL,
            cow_fk INTEGER NOT NULL,
            FOREIGN KEY(cow_fk) REFERENCES Cow(id)
        );
    `);
};

export const deleteDatabase = async () => {
  await db.closeAsync();
  await SQLite.deleteDatabaseAsync("ezcow.db");
};

export const closeDatabase = async () => {
  await db.closeAsync();
  db = null;
};

export const syncDatabase = async () => {
  await db.execAsync(`
        PRAGMA wal_checkpoint(FULL);
    `);
};
