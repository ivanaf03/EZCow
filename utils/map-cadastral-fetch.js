import { XMLParser } from "fast-xml-parser";

// XML example
/*
<FeatureCollection xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:gml="http://www.opengis.net/gml/3.2" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:cp="http://inspire.ec.europa.eu/schemas/cp/4.0" xmlns:gmd="http://www.isotc211.org/2005/gmd" xmlns="http://www.opengis.net/wfs/2.0" xsi:schemaLocation="http://www.opengis.net/wfs/2.0 http://schemas.opengis.net/wfs/2.0/wfs.xsd http://inspire.ec.europa.eu/schemas/cp/4.0 http://inspire.ec.europa.eu/schemas/cp/4.0/CadastralParcels.xsd" timeStamp="2025-05-20T17:34:00" numberMatched="1" numberReturned="1">
<member>
<cp:CadastralParcel gml:id="ES.SDGC.CP.27018A02100015">
<cp:areaValue uom="m2">7140</cp:areaValue>
<cp:beginLifespanVersion>2011-03-23T00:00:00</cp:beginLifespanVersion>
<cp:endLifespanVersion xsi:nil="true" nilReason="http://inspire.ec.europa.eu/codelist/VoidReasonValue/Unpopulated"/>
<cp:geometry>
<gml:MultiSurface gml:id="MultiSurface_ES.SDGC.CP.27018A02100015" srsName="http://www.opengis.net/def/crs/EPSG/0/25830">
<gml:surfaceMember>
<gml:Surface gml:id="Surface_ES.SDGC.CP.27018A02100015.1" srsName="http://www.opengis.net/def/crs/EPSG/0/25830">
<gml:patches>
<gml:PolygonPatch>
<gml:exterior>
<gml:LinearRing>
<gml:posList srsDimension="2" count="28">162070.81 4795525.35 162071.2 4795526.64 162073.26 4795533.33 162074.99 4795538.97 162079.48 4795547.47 162087.24 4795560.3 162091.77 4795566.22 162096.42 4795569.58 162116.86 4795582.82 162131.88 4795590.73 162160.49 4795575.17 162164.9 4795571.72 162182.37 4795557.42 162188.52 4795551.85 162180.12 4795544.61 162173.84 4795536.29 162164.43 4795524.7 162142.85 4795502.11 162133.82 4795501.56 162122.48 4795500.88 162116.68 4795494.26 162097.16 4795485.67 162092.02 4795510.92 162081.95 4795516.52 162077.57 4795502.69 162067.34 4795506.73 162069.53 4795521.18 162070.81 4795525.35</gml:posList>
</gml:LinearRing>
</gml:exterior>
</gml:PolygonPatch>
</gml:patches>
</gml:Surface>
</gml:surfaceMember>
</gml:MultiSurface>
</cp:geometry>
<cp:inspireId>
<Identifier xmlns="http://inspire.ec.europa.eu/schemas/base/3.3">
<localId>27018A02100015</localId>
<namespace>ES.SDGC.CP</namespace>
</Identifier>
</cp:inspireId>
<cp:label>00015</cp:label>
<cp:nationalCadastralReference>27018A02100015</cp:nationalCadastralReference>
<cp:referencePoint>
<gml:Point gml:id="ReferencePoint_ES.SDGC.CP.27018A02100015" srsName="http://www.opengis.net/def/crs/EPSG/0/25830">
<gml:pos>162124.53 4795540</gml:pos>
</gml:Point>
</cp:referencePoint>
</cp:CadastralParcel>
</member>
</FeatureCollection>
*/

export const fetchCadastral = async (cadastralReference) => {
  const URL = `https://ovc.catastro.meh.es/INSPIRE/wfsCP.aspx?service=wfs&version=2&request=getfeature&STOREDQUERIE_ID=GetParcel&refcat=${cadastralReference}&srsname=EPSG:4326`;

  const response = await fetch(URL);

  const xml = await response.text();
  const parser = new XMLParser();
  const parsed = parser.parse(xml);

  const cadastralParcel = parsed.FeatureCollection.member["cp:CadastralParcel"];

  const posList =
    cadastralParcel["cp:geometry"]["gml:MultiSurface"]["gml:surfaceMember"][
      "gml:Surface"
    ]["gml:patches"]["gml:PolygonPatch"]["gml:exterior"]["gml:LinearRing"][
      "gml:posList"
    ];

  const coordsArray = posList.trim().split(/\s+/);

  const polygonCoords = [];
  for (let i = 0; i < coordsArray.length; i += 2) {
    polygonCoords.push({
      latitude: parseFloat(coordsArray[i]), 
      longitude: parseFloat(coordsArray[i + 1]),
    });
  }

  return polygonCoords;
};

