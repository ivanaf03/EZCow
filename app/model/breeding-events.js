import { getDatabase } from "./bd";

export const insertBreedingEvent = async (
  cow_fk,
  description,
  date,
  user_fk
) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `INSERT INTO BreedingEvent (cow_fk, description, date, user_fk) VALUES (?, ?, ?, ?)`,
      [cow_fk, description, date, user_fk]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getBreedingEventsByDayAnUserId = async (day, userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM BreedingEvent JOIN Cow ON BreedingEvent.cow_fk = Cow.id WHERE date = ? AND user_fk = ?`,
      [userId, day]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};