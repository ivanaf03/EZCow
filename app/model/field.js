import { getDatabase } from "./bd";

export const insertField = async (name, latitude, longitude, user_fk) => {
  const db = await getDatabase();
  try {
    await db.runAsync(
      `INSERT INTO Field (name, latitude, longitude, user_fk) VALUES (?, ?, ?, ?)`,
      [name, latitude, longitude, user_fk]
    );
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllFieldsByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(`SELECT * FROM Field WHERE user_fk = ?`, [
      userId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
