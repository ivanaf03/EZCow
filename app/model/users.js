import CryptoJS from "crypto-js";
import { getDatabase } from "./bd";

const hashPassword = (password) => {
  return CryptoJS.SHA256(password).toString();
};

export const insertUser = async (name, email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `INSERT INTO User (name, email, password) VALUES (?, ?, ?)`,
      [name, email, hashedPassword]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const insertUserGoogle = async (name, email, google_id) => {
  const db = await getDatabase();
  try {
    const existingUser = await db.getFirstAsync(
      `SELECT * FROM User WHERE email = ?`,
      [email]
    );
    if (existingUser) {
      return existingUser;
    }
    const res = await db.runAsync(
      `INSERT INTO User (name, email, google_id) VALUES (?, ?, ?)`,
      [name, email, google_id]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getUserByEmail = async (email) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(`SELECT * FROM User WHERE email = ?`, [
      email,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const changePassword = async (email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `UPDATE User SET password = ? WHERE email = ?`,
      [hashedPassword, email]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
