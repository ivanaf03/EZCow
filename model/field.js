import { getDatabase } from "../model/bd";

export const insertField = async (name, latitude, longitude, user_fk) => {
  const db = await getDatabase();
  try {
    await db.runAsync(
      `INSERT INTO Field (name, latitude, longitude, user_fk) VALUES (?, ?, ?, ?)`,
      [name, latitude, longitude, user_fk]
    );
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllFieldsByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(`SELECT * FROM Field WHERE user_fk = ?`, [
      userId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const deleteFieldById = async (id) => {
  const db = await getDatabase();
  try {
    await db.runAsync(`DELETE FROM Field WHERE id = ?`, [id]);
  } catch (error) {
    console.log(error);
    return null;
  }
};
