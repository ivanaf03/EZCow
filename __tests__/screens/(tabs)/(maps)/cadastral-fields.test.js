import React from "react";

import { render, waitFor, fireEvent } from "@testing-library/react-native";
import { router } from "expo-router";

import CadastralFields from "../../../../app/(tabs)/(maps)/cadastral-fields";
import { getAllFieldsByUserId } from "../../../../model/field";
import { getUserCoordinatesById } from "../../../../model/users";
import { fetchCadastral } from "../../../../utils/map-cadastral-fetch";

jest.mock("../../../../store/user-provider", () => ({
  useUser: jest.fn(() => ({
    user: { id: "1", name: "TestUser", email: "test@test.com" },
  })),
}));

jest.mock("react-native-maps", () => {
  const React = require("react");
  const { View } = require("react-native");

  return {
    __esModule: true,
    default: (props) => React.createElement(View, props, props.children),
    Marker: (props) => React.createElement(View, props, props.children),
    Polygon: (props) => React.createElement(View, props, props.children),
  };
});

jest.mock("../../../../model/field", () => ({
  getAllFieldsByUserId: jest.fn(),
  deleteFieldById: jest.fn(),
}));

jest.mock("../../../../model/users", () => ({
  getUserCoordinatesById: jest.fn(),
}));

jest.mock("../../../../utils/map-cadastral-fetch", () => ({
  fetchCadastral: jest.fn(),
}));

jest.mock("expo-router", () => ({
  router: {
    replace: jest.fn(),
    push: jest.fn(),
  },
}));

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: () => null,
}));

describe("CadastralFields", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", async () => {
    getAllFieldsByUserId.mockResolvedValue([
      {
        id: 1,
        name: "TestField",
        cadastralReference: "12345678",
        latitude: 0,
        longitude: 0,
      },
    ]);
    getUserCoordinatesById.mockResolvedValue({ latitude: 0, longitude: 0 });
    fetchCadastral.mockResolvedValue([{ latitude: 0, longitude: 0 }]);
    const tree = render(<CadastralFields />);
    await waitFor(() => expect(tree).toMatchSnapshot());
  });

  it("should navigate to fields", async () => {
    getAllFieldsByUserId.mockResolvedValue([
      {
        id: 1,
        name: "TestField",
        cadastralReference: "12345678",
        latitude: 0,
        longitude: 0,
      },
    ]);
    getUserCoordinatesById.mockResolvedValue({ latitude: 0, longitude: 0 });
    fetchCadastral.mockResolvedValue([{ latitude: 0, longitude: 0 }]);
    const tree = render(<CadastralFields />);
    await waitFor(() => expect(tree).toMatchSnapshot());
    const fieldsButton = tree.getByTestId("fields-button");
    fireEvent.press(fieldsButton);
    await waitFor(() => expect(router.replace).toHaveBeenCalledWith("fields"));
  });

  it("should navigate to grazing", async () => {
    getAllFieldsByUserId.mockResolvedValue([
      {
        id: 1,
        name: "TestField",
        cadastralReference: "12345678",
        latitude: 0,
        longitude: 0,
      },
    ]);
    getUserCoordinatesById.mockResolvedValue({ latitude: 0, longitude: 0 });
    fetchCadastral.mockResolvedValue([{ latitude: 0, longitude: 0 }]);
    const tree = render(<CadastralFields />);
    await waitFor(() => expect(tree).toMatchSnapshot());
    const grazingButton = tree.getByTestId("grazing-button");
    fireEvent.press(grazingButton);
    await waitFor(() => expect(router.replace).toHaveBeenCalledWith("grazing"));
  });

  it("should navigate to add-fields", async () => {
    getAllFieldsByUserId.mockResolvedValue([
      {
        id: 1,
        name: "TestField",
        cadastralReference: "12345678",
        latitude: 0,
        longitude: 0,
      },
    ]);
    getUserCoordinatesById.mockResolvedValue({ latitude: 0, longitude: 0 });
    fetchCadastral.mockResolvedValue([{ latitude: 0, longitude: 0 }]);
    const tree = render(<CadastralFields />);
    await waitFor(() => expect(tree).toMatchSnapshot());
    const addFieldsButton = tree.getByText("AÃ±adir con referencia catastral");
    fireEvent.press(addFieldsButton);
    await waitFor(() =>
      expect(router.push).toHaveBeenCalledWith("add-fields-cadastral")
    );
  });
});
