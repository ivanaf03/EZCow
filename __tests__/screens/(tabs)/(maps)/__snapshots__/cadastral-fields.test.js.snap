// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CadastralFields should navigate to add-fields 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "fontFamily": "Nunito-Bold",
        },
        {
          "borderBottomWidth": 2,
        },
        {
          "borderBottomColor": "#ffffff",
          "borderLeftColor": "#ffffff",
          "borderRightColor": "#ffffff",
          "borderTopColor": "#ffffff",
        },
      ]
    }
  >
    Mis fincas (catastro)
  </Text>
  <View
    style={
      [
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "marginTop": 16,
        },
      ]
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomWidth": 2,
              "borderLeftWidth": 2,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
            },
            {
              "borderBottomColor": "#6bc0eb",
              "borderLeftColor": "#6bc0eb",
              "borderRightColor": "#6bc0eb",
              "borderTopColor": "#6bc0eb",
            },
            {
              "borderBottomLeftRadius": 24,
              "borderBottomRightRadius": 24,
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Nunito-SemiBold",
              },
              {
                "color": "#ffffff",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              [
                {
                  "marginLeft": 16,
                },
              ],
            ]
          }
        >
          AÃ±adir con referencia catastral
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginTop": 8,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-between",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomWidth": 2,
              "borderLeftWidth": 2,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
            },
            {
              "borderBottomColor": "#6bc0eb",
              "borderLeftColor": "#6bc0eb",
              "borderRightColor": "#6bc0eb",
              "borderTopColor": "#6bc0eb",
            },
            {
              "borderBottomLeftRadius": 24,
              "borderBottomRightRadius": 24,
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
          ]
        }
        testID="fields-button"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Nunito-SemiBold",
              },
              {
                "color": "#ffffff",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              {
                "textAlign": "center",
              },
            ]
          }
        >
          Ver ubicaciones
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 8,
          "marginRight": 8,
        },
      ]
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      showsVerticalScrollIndicator={true}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`CadastralFields should navigate to fields 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "fontFamily": "Nunito-Bold",
        },
        {
          "borderBottomWidth": 2,
        },
        {
          "borderBottomColor": "#ffffff",
          "borderLeftColor": "#ffffff",
          "borderRightColor": "#ffffff",
          "borderTopColor": "#ffffff",
        },
      ]
    }
  >
    Mis fincas (catastro)
  </Text>
  <View
    style={
      [
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "marginTop": 16,
        },
      ]
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomWidth": 2,
              "borderLeftWidth": 2,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
            },
            {
              "borderBottomColor": "#6bc0eb",
              "borderLeftColor": "#6bc0eb",
              "borderRightColor": "#6bc0eb",
              "borderTopColor": "#6bc0eb",
            },
            {
              "borderBottomLeftRadius": 24,
              "borderBottomRightRadius": 24,
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Nunito-SemiBold",
              },
              {
                "color": "#ffffff",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              [
                {
                  "marginLeft": 16,
                },
              ],
            ]
          }
        >
          AÃ±adir con referencia catastral
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginTop": 8,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-between",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomWidth": 2,
              "borderLeftWidth": 2,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
            },
            {
              "borderBottomColor": "#6bc0eb",
              "borderLeftColor": "#6bc0eb",
              "borderRightColor": "#6bc0eb",
              "borderTopColor": "#6bc0eb",
            },
            {
              "borderBottomLeftRadius": 24,
              "borderBottomRightRadius": 24,
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
          ]
        }
        testID="fields-button"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Nunito-SemiBold",
              },
              {
                "color": "#ffffff",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              {
                "textAlign": "center",
              },
            ]
          }
        >
          Ver ubicaciones
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 8,
          "marginRight": 8,
        },
      ]
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      showsVerticalScrollIndicator={true}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`CadastralFields should render correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "fontFamily": "Nunito-Bold",
        },
        {
          "borderBottomWidth": 2,
        },
        {
          "borderBottomColor": "#ffffff",
          "borderLeftColor": "#ffffff",
          "borderRightColor": "#ffffff",
          "borderTopColor": "#ffffff",
        },
      ]
    }
  >
    Mis fincas (catastro)
  </Text>
  <View
    style={
      [
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "marginTop": 16,
        },
      ]
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomWidth": 2,
              "borderLeftWidth": 2,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
            },
            {
              "borderBottomColor": "#6bc0eb",
              "borderLeftColor": "#6bc0eb",
              "borderRightColor": "#6bc0eb",
              "borderTopColor": "#6bc0eb",
            },
            {
              "borderBottomLeftRadius": 24,
              "borderBottomRightRadius": 24,
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Nunito-SemiBold",
              },
              {
                "color": "#ffffff",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              [
                {
                  "marginLeft": 16,
                },
              ],
            ]
          }
        >
          AÃ±adir con referencia catastral
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginTop": 8,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "space-between",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomWidth": 2,
              "borderLeftWidth": 2,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
            },
            {
              "borderBottomColor": "#6bc0eb",
              "borderLeftColor": "#6bc0eb",
              "borderRightColor": "#6bc0eb",
              "borderTopColor": "#6bc0eb",
            },
            {
              "borderBottomLeftRadius": 24,
              "borderBottomRightRadius": 24,
              "borderTopLeftRadius": 24,
              "borderTopRightRadius": 24,
            },
          ]
        }
        testID="fields-button"
      >
        <Text
          style={
            [
              {
                "fontFamily": "Nunito-SemiBold",
              },
              {
                "color": "#ffffff",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              {
                "textAlign": "center",
              },
            ]
          }
        >
          Ver ubicaciones
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 8,
          "marginRight": 8,
        },
      ]
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      showsVerticalScrollIndicator={true}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
